Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift

```diff
@@ -461,40 +465,20 @@ public final class PackageInfoMapper: PackageInfoMapping {
         switch target.type {
         case .macro, .executable:
             destinations = Set([.mac])
+        case .test:
+            var testDestinations = Set(XcodeGraph.Destination.allCases)
+            let dependencyNames = target.dependencies.map(\.name)
+            for dependencyName in dependencyNames {
+                let dependencyProducts = targetToProducts[dependencyName] ?? Set()
+                let destinations = unionDestinationsOfProducts(dependencyProducts, in: productDestinations)
+                testDestinations.formIntersection(destinations)
+            }
+            destinations = ProjectDescription.Destinations.from(destinations: testDestinations)
```

Comment:
Here we are adding test targets to the tuist projects also for external SPM dependencies, isn't it?
I think we shouldn't do it

Reply:
I tried this locally and I didn't get test targets for external dependencies. Can you double check @woin2ee?

Reply:
I didn't get either. Test targets of external SPM dependencies are ignored at the first of this method and I think it works well. I could have added the same logic here, but I chose to clarify it with comment of first.